pkg/apimachinery/srv/http_server.go

```diff
@@ -0,0 +1,128 @@
+package srv
+
+import (
+	"context"
+	"crypto/tls"
+	"errors"
+	"net/http"
+	"time"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/services/featuremgmt"
+	"github.com/grafana/grafana/pkg/setting"
+)
+
+// HTTPServerConfig is the config for the Grafana K8S HTTP server.
+type HTTPServerConfig struct {
+	Enabled         bool
+	ListenAddress   string
+	ServerCert      tls.Certificate
+	ReadTimeout     time.Duration
+	WriteTimeout    time.Duration
+	ShutdownTimeout time.Duration
+}
+
+// NewHTTPServerConfig parses and returns a new HTTPServerConfig from setting.Cfg.
+func NewHTTPServerConfig(cfg *setting.Cfg, featureEnabled bool) (HTTPServerConfig, error) {
+	if !featureEnabled {
+		return HTTPServerConfig{
+			Enabled: false,
+		}, nil
+	}
+
+	sec := cfg.Raw.Section("apiserver.server")
+
+	serverCert, err := tls.LoadX509KeyPair(
+		sec.Key("cert_file_path").MustString(""),
+		sec.Key("key_file_path").MustString(""),
+	)
+	if err != nil {
+		return HTTPServerConfig{}, err
+	}
+
+	c := HTTPServerConfig{}
+	c.Enabled = featureEnabled
+	c.ListenAddress = sec.Key("listen_address").MustString("127.0.0.1:8443")
```

Comment:
Yes, different port. And yes, definitely possible to do - apiservers support running with a prefix. 

I don't have a particularly strong opinion on running with a prefix vs. a separate port. The main benefit to a separate port is that operationally, any basic firewall can provide layer 4 assurance that the new api isn't world-writable. If there ends up being any period of time in which we have a k8s api, but haven't worked out RBAC for it, that might be especially important.

Reply:
I'd rather start with a separate port. Firstly because of what sam said regarding FW rules & security and secondly to make sure that there are absolutely no potential name clashes.